packages/store/src/workspace/addon/blob.ts

```diff
@@ -69,6 +67,26 @@ export const blob = addOnFactory<keyof BlobAddon>(
             );
             return Array.from(keys);
           },
+          gc: async () => {
+            const blobs = await this.blob.list();
+            blobs.forEach(blobId => {
+              const ref = this._blobsRef.get(blobId);
+              if (!ref || ref <= 0) {
+                this.blob.delete(blobId);
+                this._blobsRef.delete(blobId);
+              }
+            });
+          },
+          increaseRef: async blobId => {
+            const ref = this._blobsRef.get(blobId) ?? 0;
+            this._blobsRef.set(blobId, ref + 1);
+          },
+          decreaseRef: async blobId => {
+            const ref = this._blobsRef.get(blobId) ?? 0;
+            this._blobsRef.set(blobId, ref - 1 < 0 ? 0 : ref - 1);
+
+            this.blob.gc();
```

Comment:
We should delay deleting the blob, otherwise, it will break the cut or undo/redo operation.

https://github.com/toeverything/blocksuite/assets/18554747/bf77f623-9368-4d30-a5c3-c58ce403c6b2